internal/planning/deploy/wait.go

```diff
@@ -164,19 +170,21 @@ func observeContainers(ctx context.Context, env cfg.Context, cluster runtime.Clu
 						}
 
 						cws := &runtimepb.ContainerWaitStatus{}
-						if err := w.Opaque.UnmarshalTo(cws); err != nil {
-							continue
-						}
-						pending[ev.ResourceId] = append(pending[ev.ResourceId], cws)
-
-						var ctrs []*runtimepb.ContainerUnitWaitStatus
-						ctrs = append(ctrs, cws.Containers...)
-						ctrs = append(ctrs, cws.Initializers...)
-
-						for _, ctr := range ctrs {
-							if ctr.Status.Crashed || ctr.Status.Failed() {
-								failed = true
+						waitErr := &runtimepb.WaitError{}
+						if err := w.Opaque.UnmarshalTo(cws); err == nil {
```

Comment:
This feels common enough that could be just lifted into Event_WaitStatus? I.e. a new field under it, rather than going through Opaque.

Reply:
SG!